src/library/scala/collection/immutable/Vector.scala

```diff
@@ -2389,12 +2389,14 @@ private final class NewVectorIterator[A](v: Vector[A], private[this] var totalLe
   }
 
   override def slice(from: Int, until: Int): Iterator[A] = {
-    val _until =
+    val _until = until max 0
+
+    val n =
       if(from > 0) {
         drop(from)
-        until - from
-      } else until
-    take(_until)
+        _until - from
+      } else _until
+    take(n)
```

Comment:
```scala
if (util <= 0 || util < from)  this else drop(from).take(util - from)
```

Reply:
I'm unsure about your comment. Is this a third proposal? If so, I think it should be `until` instead of `util`, and there are some formatting and correctness issues (for instance, `.slice(0,0)` returns the entire vector, so `this` would need to be e.g. `Iterator.empty`).

Reply:
@ansvonwa yes, I edit it on my phone, but I think you can update your pr to avoid the usage of `max`. and @Ichoran's proposal is very direct too.